[maj0-0] iP

maj0-0 opened the pull request:
## Misty

> “I forgot.” – Me

[source](#)

Misty helps keep track of tasks you need to do. It's:

- text-based
- easy to learn
- ~~slow~~ **somewhat fast** to use
- _named after my cat :)_

### How to use:

1. download it from [here](#).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you. 😺

After downloading, run the `Main` class to start Misty.

### Features:

- [x] Managing tasks
- [x] Categorising tasks as either a Todo, Deadline, or Event
- [x] Searching for a task by it's description
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

Commit:
Add Gradle support

maj0-0 committed:
Complete Level 0 Implementation

maj0-0 committed:
Complete Level 1 Implementation

maj0-0 committed:
Complete Level 2 Implementation

maj0-0 committed:
Complete Level 3 Implementation

maj0-0 committed:
Complete Level 4 Implementation

maj0-0 committed:
Complete A-TextUiTesting Implementation

maj0-0 committed:
Complete Level 5 Implementation

maj0-0 committed:
Complete Level 6 Implementation

maj0-0 committed:
Complete Level 7 Implementation

maj0-0 committed:
Complete Level 8 Implementation

maj0-0 committed:
Create Task Interfaces Completable and Describable

maj0-0 committed:
Implement Enums for Task Types

shishirbychapur commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,202 @@
+import java.util.Scanner;
+import java.util.ArrayList;
+import java.io.File;
+import java.io.FileWriter;
```
Import statements are not following java coding standard.(should be in lexographic order)

shishirbychapur commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,202 @@
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
 public class Duke {
```
Missing javadocs for class. Most of your methods are private though, so they don't need javadocs 👍

shishirbychapur commented on src/main/java/DukeException.java:
Missing javadocs!

shishirbychapur commented on src/main/java/Event.java:
Missing javadocs!

shishirbychapur commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,84 @@
+
+public class Task {
+    protected String taskName;
+    protected boolean completed;
```
boolean variable shud be isCompleted, to follow java coding standards.

shishirbychapur commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,84 @@
+            task = null;
+        }
+
+        if (task != null && isCompleted.equals("1")){
```
there should be space between parenthesis and curly brace.

shishirbychapur commented on src/main/java/Task.java:
javadocs missing!

shishirbychapur commented on src/main/java/Todo.java:
javadocs missing!

shishirbychapur reviewed (commented):
Overall, code is formatted well, just a few nits. Javadocs missing and minor code indentation issues. Well done!

dickongwd commented on data/duke.txt:
```diff
@@ -0,0 +1,2 @@
+D | 1 | Submit report | 2023-09-01 1800
+E | 0 | Team meeting | 2023-01-01 1300 | 2023-09-02 1200
```
Consider adding the file to `.gitignore`. The data file is not relevant to be pushed to the respository!

dickongwd commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,22 @@
+        this.by = by;
+    }
+
+    public LocalDateTime getBy() {
+        return by;
+    }
```
`LocalDateTime` is an object, and when you have a getter for the object other code can access it and change it! Just a caution.

dickongwd commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,202 @@
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
 public class Duke {
+    private static ArrayList<Task> currList = new ArrayList<>();
+    private static final String FILE_PATH = "./data/duke.txt";
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
-    }
-}
+        greetUser();
+
+        Scanner sc = new Scanner(System.in);
+        while (true) {
+            try {
+                String currInput = sc.nextLine().trim();
+                processInput(currInput);
+                saveTasksToFile();
+            } catch (DukeException e) {
+                System.out.println(">  " + e.getMessage());
+            } catch (NumberFormatException e) {
+                System.out.println(">  Please provide a valid task number.");
+            }
+            System.out.println("------------------------------------------------");
+        }
+    }
```
Your `main` function is concise and easy to understand! You can consider putting the error message for `NumberFormatException` in the exception object instead of hardcoding the message.

dickongwd commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,84 @@
+        completed = false;
+    }
+
+    public String toFileString() {
+        if (this instanceof Todo) {
+            return "T | " + (completed ? "1" : "0") + " | " + taskName;
+        } else if (this instanceof Deadline) {
+            Deadline deadline = (Deadline) this;
+            return "D | " + (completed ? "1" : "0") + " | " + taskName + " | " + deadline.getBy().format(DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm"));
+        } else if (this instanceof Event) {
+            Event event = (Event) this;
+            return "E | " + (completed ? "1" : "0") + " | " + taskName + " | " + event.getFrom().format(DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm")) + " | " + event.getTo().format(DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm"));
+        } else {
+            return "";
+        }
+    }
```
Consider abstracting this to each task class instead!

dickongwd reviewed (commented):
LGTM! Just some small suggestions and nitpicks.

maj0-0 committed:
Complete A-MoreOOP Implementation

maj0-0 committed:
Complete A-Packages Implementation

maj0-0 committed:
Merge branch 'branch-A-MoreOOP'

maj0-0 committed:
Merge branch 'branch-A-Packages'

maj0-0 committed:
Implement JUnit Tests

maj0-0 committed:
Package the App as a JAR file

maj0-0 committed:
Add JavaDoc comments

maj0-0 committed:
Tweak the code to comply with a coding standard

maj0-0 committed:
Give users a way to find a task by searching for a keyword

maj0-0 committed:
Merge branch 'branch-A-JavaDoc' into branch-A-CodingStandard

# Conflicts:
#	src/main/java/task/TaskList.java
#	src/main/java/task/TaskType.java

maj0-0 committed:
Merge branch 'branch-A-CodingStandard' into branch-Level-9

maj0-0 committed:
Allow Duke to Echo User Input

maj0-0 committed:
Complete GUI Implementation of Duke

maj0-0 committed:
Generate a new JAR file

maj0-0 committed:
Add assertions for command validation

The parser relies on string checks to determine the type of command the user wants to execute.

Using assertions will allow us to catch unintended logic errors during the development and debugging phases.

maj0-0 committed:
Ensure task loading and saving validity with assertions

The Storage class handles both loading tasks from a file and saving
tasks back to that file.

Ensuring that tasks are correctly loaded and saved without null values
or unexpected states is paramount for the user experience and data integrity.

Let's add assertions in the Storage class to verify the non-null state
of tasks and the filePath.

This provides an additional layer of checking and ensures that the
code's assumptions about data integrity hold true during runtime.

maj0-0 committed:
Add assertions to ensure correct program flow and user interactions

The Duke and UI classes are central to the application's workflow and user experience. The `Duke` class oversees the main program flow while the `UI` class is responsible for all user interactions. Ensuring the correctness and stability of these classes is paramount to the application's reliability.

Previously, despite having error handling and validation in place, these classes were not equipped with assertions to validate assumptions about the state and flow of the program. This omission could potentially lead to undetected flaws in execution or inconsistencies in the UI.

To bolster the application's resilience and clarity:

- Assertions have been added to the `Duke` class to validate the correct sequence of method calls and ensure that no unintended states arise during execution.
- In the `UI` class, assertions validate the consistency of user prompts, ensuring that messages are neither null nor unintentionally altered.

By integrating these assertions, we aim to catch and rectify potential anomalies at the earliest stages of their occurrence. This proactive measure will not only enhance the user experience by preventing unforeseen disruptions but also provide developers with immediate feedback should any inconsistency arise.

maj0-0 committed:
Improve the code quality where necessary

maj0-0 committed:
Merge branch 'branch-A-Assertions' into branch-A-CodeQuality

maj0-0 committed:
Improve the GUI to look more appealing

maj0-0 committed:
Update functionalities to work with GUI

maj0-0 committed:
Provide a way to view tasks in the form of a schedule

Add implementation to view the schedule for a specific date.

maj0-0 committed:
Provide a way to view tasks in the form of a schedule

Implement functionalities to view the schedule for a specific date.

maj0-0 committed:
Merge pull request #2 from maj0-0/branch-Extension

Provide a way to view tasks in the form of a schedule

maj0-0 committed:
Fix Schedule Bugs

Fix bugs to when trying to view the schedule of a specific date.

maj0-0 committed:
Update README and Screenshot

maj0-0 committed:
Merge branch 'master' of https://github.com/maj0-0/ip

# Conflicts:
#	data/duke.txt
#	src/main/java/command/ViewScheduleCommand.java
#	src/main/java/parser/Parser.java

maj0-0 committed:
Fix Storage Problems

maj0-0 committed:
Fix Boolean Variable Naming
